rustls/src/dns_name.rs

```diff
@@ -0,0 +1,176 @@
+/// DNS name validation according to RFC1035, but with underscores allowed.
+use std::error::Error as StdError;
+use std::fmt;
+
+/// "Labels must be 63 characters or less."
+const MAX_LABEL_LENGTH: usize = 63;
+
+fn validate(input: &[u8]) -> Result<(), InvalidDnsNameError> {
```

Comment:
Style: I'd like to pitch you on the idea that we should order code the other way around, in a top-down way instead of bottom-up. That is, I think `MAX_LABEL_LENGTH` and `validate()` should go near the bottom of the module because we're more likely to want to refer to/change `DnsName` and `DnsNameRef`.

I do this in all my projects and most of the rustls code that I've had a substantial hand in is also like this. I think it's easier to navigate.

Reply:
I think my brain is still a bit C++-addled and I default from dependencies to dependants. Will try the "headlines then detail" style.

Reply:
I don't mind calling you out on that if you agree that it's worthwhile! Just wanted to make sure you were even interested in doing the headlines-first style.

Reply:
Please do!